caas/kubernetes/provider/rbac.go

```diff
@@ -14,6 +17,27 @@ import (
 	"github.com/juju/juju/caas"
 )
 
+type nameGetterSetter interface {
+	GetName() string
+	SetName(string)
+}
+
+func (k *kubernetesClient) getJujuRBACName(name string) string {
+	prefix := fmt.Sprintf("juju-%s-", k.namespace)
```

Comment:
I think we only want to prefix with the namespace when a non-namespaced entity. 

`appname-resourcename` for namespaced entities.
`juju-namespacename-appname-resourcename` for non-namespaced entities.